pkg/client/transport/cache_test.go

```diff
@@ -18,9 +18,39 @@ package transport
 
 import (
 	"net/http"
+	"reflect"
 	"testing"
 )
 
+func TestGet(t *testing.T) {
+	tests := []struct {
+		config            *Config
+		expectedTransport http.RoundTripper
+	}{
+		{
+			config:            &Config{},
+			expectedTransport: http.DefaultTransport,
+		},
+		{
+			config: &Config{
+				UnixSocket: "/var/run/k8s.sock",
+			},
+			expectedTransport: defaultUnixSocketTransport,
+		},
+	}
+
+	for _, tc := range tests {
+		transport, err := tlsCache.get(tc.config)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !reflect.DeepEqual(transport, tc.expectedTransport) {
+			t.Errorf("Expected %#v, got %#v", tc.expectedTransport, transport)
```

Comment:
I changed these tests. Now I check the fact whether some transport was cached or not (and whether that caching was expected).